oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/RelatedSelectField/ExternalApiModal.jsx

```diff
@@ -0,0 +1,230 @@
+  const [externalApiRecord, setExternalApiRecord] = useState({});
+  const { setFieldValue } = useFormikContext();
+  const searchApi = new InvenioSearchApi(searchConfig.searchApi);
+  const handleExternalRecordChange = (record) => {
+    setExternalApiRecord(record);
+  };
+  return (
+    <Modal open={open} onClose={onClose}>
+      <Modal.Header as="h6" className="pt-10 pb-10">
+        <Grid>
+          <Grid.Column floated="left">
+            <Header as="h2">Search results from external API</Header>
+          </Grid.Column>
+        </Grid>
+      </Modal.Header>
+      <Modal.Content>
+        <OverridableContext.Provider value={overriddenComponents}>
+          <ReactSearchKit
+            searchApi={searchApi}
+            initialQueryState={searchConfig.initialQueryState}
+            urlHandlerApi={{ enabled: false }}
+          >
+            <Grid celled="internally">
+              <Grid.Row>
+                <Grid.Column width={8} floated="left" verticalAlign="middle">
+                  <SearchBar
+                    placeholder={i18next.t("Search")}
+                    autofocus
+                    actionProps={{
+                      icon: "search",
+                      content: null,
+                      className: "search",
+                    }}
+                  />
+                </Grid.Column>
+              </Grid.Row>
+              <Grid.Row verticalAlign="middle">
+                <Grid.Column>
+                  <ResultsLoader>
+                    <EmptyResults />
+                    <Error />
+                    <ExternalApiResultsList
+                      handleAddingExternalApiSuggestion={
+                        handleAddingExternalApiSuggestion
+                      }
+                      fieldPath={fieldPath}
+                      handleExternalRecordChange={handleExternalRecordChange}
+                      externalApiRecord={externalApiRecord}
+                      serializeSuggestions={serializeSuggestions}
+                    />
+                  </ResultsLoader>
+                </Grid.Column>
+              </Grid.Row>
+              <Grid.Row verticalAlign="middle">
+                <Grid.Column>
+                  <Pagination options={{ size: "tiny" }} />
+                </Grid.Column>
+
+                <Grid.Column floated="right" width={3}>
+                  <ResultsPerPage
+                    values={searchConfig.paginationOptions.resultsPerPage}
+                    label={resultsPerPageLabel}
+                  />
+                </Grid.Column>
+              </Grid.Row>
+            </Grid>
+          </ReactSearchKit>
+        </OverridableContext.Provider>
+      </Modal.Content>
+      <Modal.Actions>
+        <Button
+          name="cancel"
+          onClick={() => {
+            onClose();
+          }}
+          icon="remove"
+          labelPosition="left"
+          content={i18next.t("Cancel")}
+          floated="left"
+        />
+        <Button
```

Comment:
Can we make this Button optional (only show if the whole modal & relatedseleectfield allows for select-multiple items) and trigger the choose action (setFieldValue) when user clicks on some item from the list instead?

Reply:
@mirekys Ahoj Miro, Currently, the component does not support multiple selection from external API. I will fix this.

Are we sure about the behavior that we want item to get automatically selected immediatelly when user clicks on an item in the modal (in case of single selection mode)? 
The reason I ask is because, for example user missclicks, and now he needs to make search again, go to external API modal, and look again (which could be a bit frustrating) + this will cause more API calls to the 3rd party API.